src/RtAudioInterface.cpp

```diff
@@ -170,12 +196,17 @@ void RtAudioInterface::setup(bool verbose)
     auto dev_info_input  = rtAudioIn->getDeviceInfo(index_in);
     auto dev_info_output = rtAudioOut->getDeviceInfo(index_out);
 
-    if (static_cast<unsigned int>(getNumInputChannels()) > dev_info_input.inputChannels) {
-        setNumInputChannels(dev_info_input.inputChannels);
+    if (static_cast<unsigned int>(baseChanIn) + static_cast<unsigned int>(chansIn)
+        > dev_info_input.inputChannels) {
+        baseChanIn = 0;
+        chansIn    = 2;
+        if (static_cast<unsigned int>(dev_info_input.inputChannels) < 2) {
+            chansIn = 1;
+        }
```

Comment:
The first one looks like dev_info_input.inputChannels is unsigned. So the static_cast the second time  is not necessary. Did you try making baseChanIn, chansIn and chansOut unsigned in the first place?

Reply:
Yes that turns out to be cleaner